cdap-runtime-spi/src/main/java/io/cdap/cdap/runtime/spi/runtimejob/RuntimeJobDetail.java

```diff
@@ -26,10 +26,12 @@
 public class RuntimeJobDetail {
   private final ProgramRunInfo runInfo;
   private final RuntimeJobStatus status;
+  private final RuntimeJobStatusDetails jobStatusDetails;
 
-  public RuntimeJobDetail(ProgramRunInfo runInfo, RuntimeJobStatus status) {
+  public RuntimeJobDetail(ProgramRunInfo runInfo, RuntimeJobStatus status, RuntimeJobStatusDetails jobStatusDetails) {
```

Comment:
Seems like the `jobStatusDetails` is nullable? Also, it's better to overload the constructor so that we don't need to pass in `null` in multiple places.

Reply:
Added constructor overload.